cf-agent/verify_files_utils.c

```diff
@@ -3596,13 +3596,13 @@ bool VerifyOwner(EvalContext *ctx, const char *file, const Promise *pp, Attribut
 
     if (attr.perms.groups->next == NULL && attr.perms.groups->gid == CF_UNKNOWN_GROUP)  // Only one non.existent item
     {
-        cfPS(ctx, LOG_LEVEL_ERR, PROMISE_RESULT_FAIL, pp, attr, "Unable to make file belong to an unknown group");
+        cfPS(ctx, LOG_LEVEL_ERR, PROMISE_RESULT_FAIL, pp, attr, "Unable to make file belong to an unknown group '%s'", attr.perms.groups->gid );
         *result = PromiseResultUpdate(*result, PROMISE_RESULT_FAIL);
```

Comment:
OK I understand now.

This is *not* the right place to emit specific user or group info, just that none worked. I do have access to the file name though, so I propose this:

```
root@nickanderson-ThinkPad-W550s:/tmp/cfe-2432/bin# ./cf-agent -Kf ./t.cf   error: None of the promised groups for '/tmp/testfile' exist. Unable to assign ownership to a group that does not exist. See INFO or VERBOSE logs for more info.
   error: None of the promised groups for '/tmp/testfile' exist. Unable to assign ownership to a group that does not exist. See INFO or VERBOSE logs for more info.
```